packages/shared/src/components/CombatReport/CombatVideo/VideoPlayerTimeline.tsx

```diff
@@ -0,0 +1,107 @@
+import { CombatUnitType } from '@wowarenalogs/parser';
+import _ from 'lodash';
+import moment from 'moment';
+import { useCallback, useEffect, useMemo } from 'react';
+import { FaSkullCrossbones } from 'react-icons/fa';
+import { TbPlayerPause, TbPlayerPlay } from 'react-icons/tb';
+
+import { useCombatReportContext } from '../CombatReportContext';
+import { useVideoPlayerContext } from './VideoPlayerContext';
+
+export const VideoPlayerTimeline = () => {
+  const { combat } = useCombatReportContext();
+  const { playState, userInputs, combatTime } = useVideoPlayerContext();
+
+  const players = useMemo(() => {
+    return _.values(combat ? combat.units : []).filter((u) => u.type === CombatUnitType.Player);
+  }, [combat]);
+
+  const deaths = useMemo(
+    () => _.flatten(players.map((p) => p.deathRecords)).sort((a, b) => a.timestamp - b.timestamp),
+    [players],
+  );
+
+  useEffect(() => {
+    const onKey = (e: KeyboardEvent) => {
+      if (e.code === 'Space') {
+        playState === 'playing' ? userInputs.emit('pause') : userInputs.emit('play');
+      }
+    };
+    window.addEventListener('keyup', onKey);
+
+    return () => {
+      window.removeEventListener('keyup', onKey);
+    };
+  }, [playState, userInputs]);
+
+  useEffect(() => {
+    if (combat && combatTime >= combat.endTime) {
+      userInputs.emit('pause');
+    }
+  }, [combat, combatTime, userInputs]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedJump = useCallback(
+    _.debounce((time: number) => {
```

Comment:
Do we need to debounce here? Is there a race condition on the other side for 2 of these being close?